### PR TITLE
Add unpack function to archive module.

### PR DESCRIPTION
**Pull request by hacktobeer**
Add a helper function to unpack archives using 7zip. It supports a variety of archive formats by default: https://www.7-zip.org/

### CODE REVIEW

File: openrelik_worker_common/archives.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        input_path,
+        f"-o{export_folder}",
+    ]
+
+    command_string = " ".join(command)
+    print(f"command: {command}")
+    with open(log_file, "wb") as out:
+        process = subprocess.Popen(command, stdout=out, stderr=out)
+        while process.poll() is None:
+            time.sleep(2)
```
Comment (berggren) [inline openrelik_worker_common/archives.py:48]: If you don't need to track progress etc you can just make a blocking `call` instead
```suggestion
 subprocess.call(command)
```

File: openrelik_worker_common/archives.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    command = [
+        "7z",
+        "x",
+        input_path,
+        f"-o{export_folder}",
+    ]
+
+    command_string = " ".join(command)
+    print(f"command: {command}")
```
Comment (berggren) [inline openrelik_worker_common/archives.py:44]:
```suggestion
```

File: openrelik_worker_common/archives.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper methods for archives."""
+import os
+import subprocess
+import time
+from uuid import uuid4
+
+
+def unpack(input_path: str, output_folder: str, log_file: str) -> str:
```
Comment (berggren) [inline openrelik_worker_common/archives.py:21]: We will have support for more archive types, rename to reflect the archive type
```suggestion
def extract_7zip(input_path: str, output_folder: str, log_file: str) -> str:
```

**Review by berggren — CHANGES_REQUESTED** [review]

**Comment by berggren** [comment]
@hacktobeer PTAL.

Also, do we need to add 7z to the Dockerfile or is it installed by default?

**Review by hacktobeer — COMMENTED** [review]
Thanks for comments, PTAL.

Yes, we need 7zip in the container that wants to use this function. The PR in the plaso worker (adding task ExtractArchive) has 7z added to the Dockerfile, I can do the other containers after. I also added a check to the beginning of the function to see if it is available.

File: openrelik_worker_common/archives.py (hunk 3 above)
Comment (berggren) [reply openrelik_worker_common/archives.py]: You right, I'm just used to use separate tools for the archives :) Still good to name it descriptive IMO 👍🏼

**Comment by berggren** [comment]
> Thanks for comments, PTAL.
> 
> Yes, we need 7zip in the container that wants to use this function. The PR in the plaso worker (adding task ExtractArchive) has 7z added to the Dockerfile, I can do the other containers after. I also added a check to the beginning of the function to see if it is available.

Yeah, let's add it to the template later.

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**